src/value_serializer.rs

```diff
@@ -116,6 +119,8 @@ pub unsafe extern "C" fn v8__ValueSerializer__Delegate__BASE__FreeBufferMemory(
   this: &mut ValueSerializerDelegate,
   buffer: *mut c_void,
 ) {
+  let buffer =
+    Box::from_raw(std::ptr::slice_from_raw_parts_mut(buffer as *mut u8, 1));
```

Comment:
I think this works; but it is definitely not ideal. V8 does not give us the old buffer size, because normally calling free(ptr) in c does not require this. Although for Rust this seems to be required.

Reply:
Actually, I just realized that if every ValueSerializerDelegate will only be allocating one buffer (and I'm pretty sure of that, but will verify), it's really not such a big deal. The implementer just store the previously allocated length -- we could even decide to make rusty_v8 store it on `struct ValueSerializerDelegateBase`.

Reply:
Yeah, this would be a good solution. I now also understand what you told me earlier [the ReallocateBufferMemory method does not receive the "old" length of the buffer](https://github.com/denoland/rusty_v8/pull/442#discussion_r479556457). I thought the `size` parameter did supply that information, but this parameter actually indicates the **wanted** buffer size instead of the current one. So the current implementation contains a bug.

Reply:
> Actually, I just realized that if every ValueSerializerDelegate will only be allocating one buffer (and I'm pretty sure of that, but will verify), it's really not such a big deal. The implementer just store the previously allocated length -- we could even decide to make rusty_v8 store it on `struct ValueSerializerDelegateBase`.

The only problem I see is reusing delegates for multiple ValueSerializers.

Reply:
Ideally the buffer-length should be added to the `ValueSerializer` struct instead, a serializer always has only 1 buffer.
Although this requires the `ValueSerializerDelegate` to be able to access the `ValueSerializer` struct.